src/mlpack/config.hpp

```diff
@@ -75,40 +75,40 @@
 //
 // Perform autodetection of STB if possible.
 //
-#ifndef MLPACK_HAS_STB
-  #if defined __has_include
-    #if __has_include("stb_image.h")
-      #define MLPACK_HAS_STB_IMAGE
-    #elif __has_include("stb/stb_image.h")
-      #define MLPACK_HAS_STB_IMAGE
-    #endif
+//#ifndef MLPACK_HAS_STB
```

Comment:
I know you're just experimenting for now, but probably before merge we should have some macro that allows a user to bring their own STB instead of using the bundled one... I guess at the point we actually include, we would do something like:

```
#ifdef MLPACK_DONT_USE_BUNDLED_STB // probably needs a shorter name
  ... do all of the include things with __has_include that we do now ...
#else
  #include "stb_image.h"
  #include "stb_image_write.h"
#endif
```

Or something along those lines.